apps/condo/domains/marketplace/schema/Invoice.js

```diff
@@ -0,0 +1,196 @@
+
+        number: {
+            schemaDoc: 'The invoice number within organization',
+            type: AutoIncrementInteger,
+            isRequired: true,
+            autoIncrementScopeFields: ['context.organization'],
+        },
+
+        property: {
+            schemaDoc: 'The payer\'s property',
+            type: 'Relationship',
+            ref: 'Property',
+            knexOptions: { isNotNullable: false },
+            kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
+        },
+
+        unitType: {
+            schemaDoc: 'The payer\'s unitType',
+            type: 'Text',
+        },
+
+        unitName: {
+            schemaDoc: 'The payer\'s unitName',
+            type: 'Text',
+        },
+
+        accountNumber: {
+            schemaDoc: 'The payer\'s accountNumber within organization',
+            type: 'Text',
+        },
+
+        toPay: MONEY_AMOUNT_FIELD,
+
+        rows: {
+            schemaDoc: 'The list of paid items',
+            type: 'Json',
+            isRequired: true,
+            hooks: {
+                validateInput: validateRowsField,
+            },
+            extendGraphQLTypes: [rowsGqlSchemaTypes],
+            graphQLInputType: `[${INVOICE_ROW_GQL_INPUT_NAME}!]!`,
+            graphQLReturnType: `[${INVOICE_ROW_GQL_TYPE_NAME}!]!`,
+            graphQLAdminFragment: `{ ${Object.keys(invoiceRowSchemaFields).join(' ')} }`,
+        },
+
+        ticket: {
+            schemaDoc: 'The ticket related to this invoice',
+            type: 'Relationship',
+            ref: 'Ticket',
+            kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
+        },
+
+        contact: {
+            schemaDoc: 'The contact the invoice created for. Fill if invoice was created or connected via the ticket form',
+            type: 'Relationship',
+            ref: 'Contact',
+            kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
+        },
+
+        client: {
+            schemaDoc: 'The user who sees the invoice. Must filled with the user of corresponding resident.',
+            type: 'Relationship',
+            ref: 'User',
+            kmigratorOptions: { null: true, on_delete: 'models.SET_NULL' },
+        },
+
+        status: {
+            schemaDoc: 'The regime of counting taxes for company',
+            isRequired: true,
+            type: 'Select',
+            dataType: 'string',
+            options: INVOICE_STATUSES,
+            defaultValue: INVOICE_STATUS_DRAFT,
+        },
+
+    },
+    hooks: {
+        validateInput: ({ resolvedData, operation, existingItem, context }) => {
+            const nextData = { ...existingItem, ...resolvedData }
+            if (!((!!nextData.property && !!nextData.unitType && !!nextData.unitName) || !!nextData.accountNumber)) {
+                throw new GQLError(ERRORS.NO_RECEIVERS, context)
+            }
+
+            if (operation === 'update' && existingItem.status === INVOICE_STATUS_PAID) {
```

Comment:
You need to check the cash paid case. Probably, you want to uncheck it if you paid it by cash

Reply:
Test this case?

Reply:
Added a todo for this. After task with Payments.